src/modules/endpoint/endpoints/categories.js

```diff
@@ -1,53 +1,50 @@
 import CategoriesMapModel from '../models/CategoriesMapModel'
 import getCurrentCityConfig from '../../city-detection/getCurrentCityConfig'
+import moment from 'moment'
+import { toPairs } from 'lodash/object'
+import { compose } from 'lodash/fp'
+import normalizePath from 'normalize-path'
 
-function getUrlWithoutCity (url) {
-  const tempUrl = url.split('/')
-  tempUrl.shift()
-  return tempUrl.join('/')
+const PREFIX_LENGTH = 4
+
+// Removes the (unnecessary) path prefix '/:cmsname/:lang/:parent_topic',
+// e.g. '/neuburgschrobenhausenwohnraum/de/raumfrei'
+function getPathWithoutPrefix (path) {
+  if (!path) {
+    return ''
+  }
+  const splitPath = path.split('/').slice(PREFIX_LENGTH)
+  return `/${splitPath.join('/')}`
 }
 
 export default new EndpointBuilder('categories')
   .withStateToUrlMapper(state => {
     return `https://cms.integreat-app.de/${getCurrentCityConfig(state.cityConfigs._data).cmsName}` +
-      `/de/wp-json/extensions/v0/modified_content/pages?since=1970-01-01T00:00:00Z`
+      `/de/wp-json/extensions/v3/pages`
   })
   .withMapper((json, state) => {
-    const baseUrl = ``
-    const categories = json.filter(category => category.status === 'publish')
-      .map(category => {
-        return new CategoryModel({
-          id: category.id,
-          url: normalizeUrl(`${baseUrl}/${getUrlWithoutCity(category.permalink.url_page)}`),
-          title: category.title,
-          parentId: category.parent,
-          content: category.content,
-          thumbnail: category.thumbnail,
-          order: category.order
-        })
+    const normalize = compose([decodeURIComponent, normalizePath, getPathWithoutPrefix])
+    const basePath = '/'
+    const categories = json.map(category => {
+      return new CategoryModel({
+        id: category.id,
+        path: normalize(category.path),
+        title: category.title,
+        content: category.content,
+        thumbnail: category.thumbnail,
+        order: category.order,
+        availableLanguages: new Map(toPairs(category.available_languages)
+          .map(([key, value]) => [key, normalize(value.path)])),
+        parentPath: normalize(category.parent.path || basePath),
+        lastUpdate: moment(category.modified_gmt)
       })
-
-    const root = categories.find(_category => _category._url === baseUrl)
-    const rootId = root.id
-    root.setId(0)
-
-    categories.forEach(category => {
-      if (category.parentId === rootId) {
-        category.setParentId(0)
-      }
-      if (category.id !== 0) {
-        const parent = categories.find(_category => _category.id === category.parentId)
-        if (!parent) {
-          throw new Error(`Invalid data from categories endpoint: Page with id ${category.id} has no parent.`)
-        }
-        category.setParentUrl(parent.url)
-      }
     })
 
+    // Set the page with path '/' as root page
+    categories.find(categoryModel => categoryModel.path === '/').id = 0
```

Comment:
Is there always a root category delivered by the cms? Because in the webapp there is no root category delivered, we have to add it manually.

Reply:
I thought there was some kind of problem but I'll check.

Reply:
Two things:
* We show a `CategoriesList` as the root page (in the webapp it is the `Tiles`). Therefore we would need content (and others) in the root page, which would make the new, artificial root page essentially a clone of the page with path '/'. I thought it's easier to just set this page's id to 0.
* Another difference to the webapp is the `getPathWithoutPrefix` which we need to clip the path. So we can't just use the webapp's endpoint here.

Reply:
Okay, quite weird that you get the root category from the cms but okay :smile: